sdk/core/core-http/src/operationSpec.ts

```diff
@@ -43,6 +43,11 @@ export interface OperationSpec {
    */
   readonly contentType?: string;
 
+  /**
+   * The media type of the request body.
+   * This value can be used to aide in serialization if it is provided.
+   */
+  readonly mediaType?: string;
```

Comment:
would it make sense for mediaType to be a union type of the possible MediaTypes instead of a just a string. For instance

```typescript
readonly mediaType?: "application" | "text" | "audio" | ...
```

Reply:
Sure, I can create a union based off of the [KnownMediaType](https://github.com/Azure/perks/blob/master/codegen/media-types.ts#L17-L26) enum we use in autorest (but also allow `string` for forwards-compatibility)

Reply:
Updated!

Reply:
Note that this doesn't really do anything since string is a supertype of all the string literals. TypeScript sadly doesn't even offer parameter help for the known string values, though I consider this a bug.

Reply:
oops I missed this comment by skipping to the code 🤷‍♂️